src/app/screens/Receive/index.tsx

```diff
@@ -1,374 +1,187 @@
-    tCommon("actions.copy_invoice") as string
-  );
-  const [copyLightningAddressLabel, setCopyLightningAddressLabel] = useState(
-    t("actions.copy_lightning_address") as string
-  );
   const [lightningAddress, setLightningAddress] = useState("");
-  const [paid, setPaid] = useState(false);
-  const [pollingForPayment, setPollingForPayment] = useState(false);
-  const mounted = useRef(false);
   const isAlbyLNDHubUser = isAlbyLNDHubAccount(
     auth.account?.alias,
     auth.account?.connectorType
   );
   const isAlbyOAuthUser = isAlbyOAuthAccount(auth.account?.connectorType);
   const isAlbyUser = isAlbyOAuthUser || isAlbyLNDHubUser;
 
-  useEffect(() => {
-    mounted.current = true;
-
-    return () => {
-      mounted.current = false;
-    };
-  }, []);
-
-  const [fiatAmount, setFiatAmount] = useState("");
-
-  useEffect(() => {
-    if (formData.amount !== "" && showFiat) {
-      (async () => {
-        const res = await getFormattedFiat(formData.amount);
-        setFiatAmount(res);
-      })();
-    }
-  }, [formData, showFiat, getFormattedFiat]);
-
-  function handleChange(
-    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
-  ) {
-    setFormData({
-      ...formData,
-      [event.target.name]: event.target.value.trim(),
-    });
-  }
-
-  function checkPayment(paymentHash: string) {
-    setPollingForPayment(true);
-    poll({
-      fn: () =>
-        msg.request("checkPayment", { paymentHash }) as Promise<{
-          paid: boolean;
-        }>,
-      validate: (payment) => payment.paid,
-      interval: 3000,
-      maxAttempts: 20,
-      shouldStopPolling: () => !mounted.current,
-    })
-      .then(() => {
-        setPaid(true);
-        auth.fetchAccountInfo(); // Update balance.
-      })
-      .catch((err) => console.error(err))
-      .finally(() => {
-        setPollingForPayment(false);
-      });
-  }
-
-  function setDefaults() {
-    setFormData({
-      amount: "0",
-      description: "",
-      expiration: "",
-    });
-    setPaid(false);
-    setPollingForPayment(false);
-    setInvoice(null);
-  }
-
-  async function createInvoice() {
-    try {
-      setLoadingInvoice(true);
-      const response = await api.makeInvoice({
-        amount: formData.amount,
-        memo: formData.description,
-      });
-      setInvoice(response);
-      checkPayment(response.rHash);
-    } catch (e) {
-      if (e instanceof Error) {
-        toast.error(e.message);
-      }
-    } finally {
-      setLoadingInvoice(false);
-    }
-  }
-
-  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
-    event.preventDefault();
-    createInvoice();
-  }
-
   async function getLightningAddress() {
-    setLoadingLightningAddress(true);
-    const response = await api.getAccountInfo();
-    const lightningAddress = response.info.lightning_address;
+    // Wait for a minimum delay to prevent flickering
+    const [accountInfo] = await Promise.all([api.getAccountInfo(), delay(250)]);
```

Comment:
this wouldn't be needed if we use SWR I think. Currently it reloads the lightning address every time you visit this page